client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -599,8 +599,15 @@ public V call() throws Exception {
       return new MetadataValueImpl<V>(value.getCreated(), value.getLifespan(), value.getLastUsed(), value.getMaxIdle(), value.getVersion(), valueObj);
    }
 
-   private int toSeconds(long duration, TimeUnit timeUnit) {
-      return (int) timeUnit.toSeconds(duration);
+   protected static int toSeconds(long duration, TimeUnit timeUnit) {
+      int seconds = (int) timeUnit.toSeconds(duration);
+      long inverseDuration = timeUnit.convert(seconds, TimeUnit.SECONDS);
+
+      if (duration > inverseDuration) {
+         //Round up.
+         seconds++;
```

Comment:
With `Math.round(timeUnit.toMillis(duration) / 1000.0)` you'd change the result by 500ms instead of a full second. You'd need a special case for 0, though.